src/components/dashboard/ReceiveSummary.js

```diff
@@ -20,24 +22,30 @@ export type ReceiveProps = {
   theme: any,
 }
 
-const ReceiveAmount = ({ screenProps, styles, ...props }: ReceiveProps) => {
+const ReceiveAmount = ({ screenProps, styles }: ReceiveProps) => {
+  const gdStore = GDStore.useStore()
   const [screenState] = useScreenState(screenProps)
-  const { params } = props.navigation.state
+  const { canShare, generateReceiveShareObject, generateReceiveShareText } = useNativeSharing()
 
+  const { fullName } = gdStore.get('profile')
+  const { account, networkId } = goodWallet
   const { amount, reason, counterPartyDisplayName } = screenState
 
-  const { account, networkId } = goodWallet
-  const codeObj = useMemo(() => generateCode(account, networkId, amount, reason, counterPartyDisplayName), [
-    account,
-    networkId,
-    amount,
-    reason,
-    counterPartyDisplayName,
-  ])
+  const codeSource = [account, networkId, amount, reason, counterPartyDisplayName]
+  const codeObject = useMemo(() => generateCode(...codeSource), [generateCode, ...codeSource])
+
+  const shareStringSource = [codeObject, amount, counterPartyDisplayName, fullName]
```

Comment:
```
hmm as you're passing shareString via param object it would be better to memoize the whole object
e.g

const sharingParamsSource = [codeObject, amount, counterPartyDisplayName, fullName]
const sharingParams = useMemo(
     () => ({
        paymentLink: (canShare ? generateReceiveShareObject : generateReceiveShareText)(...shareStringSource))
      }),
     [...shareStringSource, canShare, generateReceiveShareObject, generateReceiveShareText]
)

...
<PushButton
  ...
  params={sharingParams}
```

Reply:
done

Reply:
pls rename to sharingParamsSource / sharingParams
this isn't a string anymore but object
so variable name should also be changed otherwise other developers or event you (some time later) could be confused in implementation